src/librustc/session/mod.rs

```diff
@@ -627,6 +628,22 @@ pub fn build_session_(sopts: config::Options,
         }
     };
     let target_cfg = config::build_target_config(&sopts, &span_diagnostic);
+
+    // Hook up the codemap with a callback that allows it to register FileMap
+    // accesses with the dependency graph.
+    let cm_depgraph = dep_graph.clone();
+    let codemap_dep_tracking_callback = Box::new(move |filemap: &FileMap| {
```

Comment:
I see why you were unhappy. 

This makes me wonder: I think that eventually we probably want some way to "hide" data in the tcx (e.g., maybe using privacy), so that it is only exposed via query. This would be a way to have stuff like random bits in the session be treated universally by the red-green system.

(For stuff that's hashed initially, like command-line-options, we might want to make some public accessor methods I guess.)